lbuild/config.py

```diff
@@ -28,7 +28,7 @@
 class ConfigNode(anytree.AnyNode):
 
     def __init__(self, parent=None):
-        anytree.AnyNode.__init__(self, parent, filename=Path())
+        anytree.AnyNode.__init__(self, parent)
```

Comment:
This changes behavior when rendering the tree, since anytree will render the passed kw's in the node automatically.
```
 $ lbuild -v discover
[INFO] lbuild.parser:
ConfigNode(filename='project.xml')
╰── ConfigNode(filename='../../../src/modm/board/disco_f469ni/board.xml')
    ╰── ConfigNode(filename='../../lbuild.xml')
```
With this change, this will be rendered as:
```
 $ lbuild -v discover                                                                                      [18:01:42]
[INFO] lbuild.parser:
ConfigNode()
╰── ConfigNode()
    ╰── ConfigNode()
```
Which isn't particularly useful for understanding config inheritance.

Reply:
This isn't an issue for `lbuild.node`, since we render the node part manually.

Reply:
Are you sure? I've just tested that, for me it works fine with the change.

Reply:
Oh, you're right, AnyNode just renders any public variable and not just the ones registered with the constructor!